src/main/scala/chess/Game.scala

```diff
@@ -62,12 +62,12 @@ case class Game(
       }
     }
 
-  def apply(uci: Uci.Move): Validated[String, (Game, Move)]  = apply(uci.orig, uci.dest, uci.promotion)
-  def apply(uci: Uci.Drop): Validated[String, (Game, Drop)]  = drop(uci.role, uci.pos)
-  def apply(uci: Uci): Validated[String, (Game, MoveOrDrop)] =
+  def apply(uci: Uci.Move): Validated[String, (Game, Move)] = apply(uci.orig, uci.dest, uci.promotion)
+  def apply(uci: Uci.Drop): Validated[String, (Game, Drop)] = drop(uci.role, uci.pos)
+  def apply(uci: Uci): Validated[String, (Game, Action)]    =
     uci match {
-      case u: Uci.Move => apply(u) map { case (g, m) => g -> Left(m) }
-      case u: Uci.Drop => apply(u) map { case (g, d) => g -> Right(d) }
+      case u: Uci.Move => apply(u) map { case (g, m) => g -> m }
+      case u: Uci.Drop => apply(u) map { case (g, d) => g -> d }
```

Comment:
these look to be doing the same thing whether its a Uci.Move or Uci.Drop. Can we not remove the match here? as the cases are the same